runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -304,13 +301,39 @@ else if (paused && !skillPaused)
 			pauseSkill.setText("Pause");
 		}
 
+		// Update information labels
 		// Update exp per hour separately, every time (not only when there's an update)
-		expHour.setText(htmlLabel("XP/Hour: ", xpSnapshotSingle.getXpPerHour()));
+		topLeftStat.setText(htmlLabel(xpTrackerConfig.xpPanelLabel1(), xpSnapshotSingle));
+		topRightStat.setText(htmlLabel(xpTrackerConfig.xpPanelLabel2(), xpSnapshotSingle));
+		bottomLeftStat.setText(htmlLabel(xpTrackerConfig.xpPanelLabel3(), xpSnapshotSingle));
+		bottomRightStat.setText(htmlLabel(xpTrackerConfig.xpPanelLabel4(), xpSnapshotSingle));
+	}
+
+	static String htmlLabel(XpPanelLabel panelLabel, XpSnapshotSingle xpSnapshotSingle)
+	{
+		String key = panelLabel.getActionKey(xpSnapshotSingle) + ": ";
+		if (panelLabel == XpPanelLabel.TIME_TO_LEVEL)
+		{
+			long remainingSeconds = xpSnapshotSingle.getSecondsTillGoal();
+			long durationHours = remainingSeconds / (60 * 60);
+			long durationMinutes = (remainingSeconds % (60 * 60)) / 60;
+			long durationSeconds = remainingSeconds % 60;
+
+			String ttl = String.format("%02d:%02d:%02d", durationHours, durationMinutes, durationSeconds);
```

Comment:
Can't you simply use getTimeToLevel? It already contains formatted string

Reply:
That was original take, but that contains the string "X days hh:mm:ss" and that string would be too long and continue on the next line. 

I chose to keep it hh:mm:ss for a fixed len string.

Reply:
Then maybe just move this to XpStateSingle and just reuse existing logic there (e.g maybe instead of adding secondsTillGoal simply add timeTillGoalShort)